applications/RomApplication/custom_strategies/global_rom_builder_and_solver.h

```diff
@@ -269,6 +269,52 @@ class GlobalROMBuilderAndSolver : public ResidualBasedBlockBuilderAndSolver<TSpa
         });
     }
 
+    void MonotonicityPreserving(
+        TSystemMatrixType& rA,
+        TSystemVectorType& rB
+    ) 
+    {
+        const auto& r_dof_set = BaseType::GetDofSet();
+        Vector dofs_values = ZeroVector(r_dof_set.size());
+
+        block_for_each(r_dof_set, [&](Dof<double>& rDof){
+            const std::size_t id = rDof.EquationId();
+            dofs_values[id] = rDof.GetSolutionStepValue();
+        });
+        double *values_vector = rA.value_data().begin();
+        std::size_t *index1_vector = rA.index1_data().begin();
+        std::size_t *index2_vector = rA.index2_data().begin();
+
+        std::size_t nonzero_counter = 0;
+
+        IndexPartition<std::size_t>(rA.size1()).for_each(
+            [&](std::size_t i)
+            {
+                for (std::size_t k = index1_vector[i]; k < index1_vector[i + 1]; k++) {
+                    const double value = values_vector[k];
+                    if (value > 0.0) {
+                        // increment the counter
+                        nonzero_counter++;
+                        const auto j = index2_vector[k];
+                        if (j > i) {
+                            rA(i,j) -= value;
```

Comment:
this is potentially problematic, since you are writing onto A(i,j) and A(j,i).
so you also have a potential writing conflict there!

Reply:
Done.